docs/types.rst

```diff
@@ -301,153 +273,141 @@ Where ``x`` is a byte array and ``_start`` as well as ``_len`` are integer value
-``slice(x, start=_start, len=_len)``  Return a slice of ``_len`` starting at ``_start``.
-====================================  ============================================================
-
-Where ``x`` is a byte array or string while ``_start`` and ``_len`` are integers.
-The ``len``, ``keccak256``, ``concat``, ``slice`` operators can be used with ``String`` and ``Bytes`` types.
-
 .. index:: !reference
 
 Reference Types
-***************
+===============
 
 Reference types do not fit into 32 bytes. Because of this, copying their value is not as feasible as
 with value types. Therefore only the location, i.e. the reference, of the data is passed.
 
 .. index:: !arrays
 
 Fixed-size Lists
-================
+----------------
 
 Fixed-size lists hold a finite number of elements which belong to a specified type.
 
 Syntax
-------
+******
 
 Lists can be declared with ``_name: _ValueType[_Integer]``. Multidimensional lists are also possible.
 
-**Example:**
-::
+.. code-block:: python
 
-  #Defining a list
-  exampleList: int128[3]
-  #Setting values
-  exampleList = [10, 11, 12]
-  exampleList[2] = 42
-  #Returning a value
-  return exampleList[0]
+    # Defining a list
+    exampleList: int128[3]
 
-.. index:: !structs
+    # Setting values
+    exampleList = [10, 11, 12]
+    exampleList[2] = 42
+
+    # Returning a value
+    return exampleList[0]
+
+.. _types-struct:
 
 Structs
-=======
+-------
 
 Structs are custom defined types that can group several variables.
 
 Syntax
-------
+******
 
-Structs can be accessed via ``struct.argname``.
-**Example:**
-::
+Struct members can be accessed via ``struct.argname``.
+
+.. code-block::
 
-  #Defining a struct
-  struct MyStruct:
-      value1: int128
-      value2: decimal
-  exampleStruct: MyStruct
-  #Constructing a struct
-  exampleStruct = MyStruct({value1: 1, value2: 2})
-  #Accessing a value
-  exampleStruct.value1 = 1
+    # Defining a struct
+    struct MyStruct:
+        value1: int128
+        value2: decimal
+
+    # Declaring a struct variable
+    exampleStruct: MyStruct = MyStruct({value1: 1, value2: 2})
+
+    # Accessing a value
+    exampleStruct.value1 = 1
 
 
 .. index:: !mapping
 
 Mappings
-========
+--------
 
-Mappings in Vyper can be seen as `hash tables <https://en.wikipedia.org/wiki/Hash_table>`_ which are virtually initialized such that
-every possible key exists and is mapped to a value whose byte-representation is
-all zeros: a type's default value. The similarity ends here, though: The key data is not actually stored
-in a mapping, only its ``keccak256`` hash used to look up the value. Because of this, mappings
-do not have a length or a concept of a key or value being "set".
+Mappings are `hash tables <https://en.wikipedia.org/wiki/Hash_table>`_ that are virtually initialized such that every possible key exists and is mapped to a value whose byte-representation is all zeros: a type's :ref:`default value <types-initial>`.
```

Comment:
Maybe mention that the default value is the same value produced by `empty(type)`

Reply:
There is a link to the default values section where `empty` is mentioned.